### PR TITLE
apply some fixes to acquire_configure

### PR DESCRIPTION
**Pull request by nclack**
While testing with Nenad, I ran across some issues:

- `video_stream_requirements_check` logic was fragile.
- On return the state was getting written improperly. Modified to call `abort` when things misconfigure.
- when a video stream is misconfigured the valid stream bit should be unset

### CODE REVIEW

File: src/acquire.c [hunk 1]
```diff
@@ -317,13 +317,14 @@ acquire_configure(struct AcquireRuntime* self_,
                 TRACE("Configured video stream %d.", istream);
             } else {
                 TRACE("Failed to configure video stream %d.", istream);
+                self->valid_video_streams &= ~(1 << istream);
```
Comment (nclack) [inline src/acquire.c:320]: Looking at this again, it doesn't look necessary to unset the bit

**Comment by aliddell** [comment]
Could you also open this against acquire-common?

File: src/acquire.c (hunk 1 above)
Comment (aliddell) [reply src/acquire.c:320]: Couldn't hurt, may be useful for inspecting later.

File: src/acquire.c [hunk 2]
```diff
@@ -317,13 +317,14 @@ acquire_configure(struct AcquireRuntime* self_,
...
             }
         }
     }
     TRACE("Valid video streams: code %#04x", self->valid_video_streams);
-    self->state = self->valid_video_streams > 0
-                    ? DeviceState_Armed
-                    : DeviceState_AwaitingConfiguration;
```
Comment (aliddell) [inline src/acquire.c:326]: Don't you want to set state to `DeviceState_Armed`?

Comment (aliddell) [reply src/acquire.c:326]: Per discussion, could be running. Still needs to go to Armed if it was AwaitingConfiguration.

**Comment by nclack** [comment]
closing in favor of acquire-project/acquire-common#36

**Pull request closed without merging**